documentation/specs/task-isolation-and-dependencies.md

```diff
@@ -0,0 +1,38 @@
+# Task isolation
+## Problem definition
+Tasks in MSBuild are dynamically loaded assemblies with potentially separate and colliding dependency trees. Currently MSBuild on .NET Core has no isolation between tasks and as such only one version of any given assembly can be loaded. Prime example of this is Newtonsoft.Json which has multiple versions, but all the tasks must agree on it to work.
```

Comment:
Is this true for all assemblies or only assemblies in msbuild.deps.json?

I thought (but could be wrong) that the msbuild assembly resolver could still load more than one version in default ALC if it was not in msbuild.deps.json / TPA.

Reply:
Multiple versions of the same assembly cannot exist in the same context.

Reply:
Oh my. I see. Things are quite a bit worse than I realized then.